### DISCUSSION TITLE
Session lost if using "Livewire::setUpdateRoute"

### DISCUSSION THREAD

**Discussion by kaju74** [livewire/livewire#7106] [7. Bugs (archived)]
### Livewire version

3.0.10

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Configure AppServiceProvider to support an own update route:

```
Livewire::setUpdateRoute(static function ($handle) {
    return Route::post('/roundtrip/update', $handle);
});
```

2. Store something in the session before redirecting:

```
public function email(): void
{
    // put selected user to session...
    session()->put('contact.user', $this->selected);

    // ...and redirect to formular
    $this->redirect(Contact::class, true);
}
```

3. In the ContactClass try to get the stored session value!

As soon as I use a custom update route, the session will lost during the calls and can't be used (except the _token value).

If I use the standard route (without overwriting it in the AppServiceProvider), everything works fine as expected!

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by serge-kilimoff** [comment]
Same problem here.
It's really complicated to work with livewire in a subfolder.

**Comment by serge-kilimoff** [comment]
I found the solution to this problem. This is lacking in the livewire documentation [here](https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint).
Looking at Livewire's code, the default route uses web middleware. Ideally this should be clarified in the documentation, I think.

```php
Livewire::setUpdateRoute(function ($handle) {
    // Here
    return Route::post('/custom/livewire/update', $handle)->middleware('web');
});
```

**Reply by justindantzer** [reply]
@serge-kilimoff old issue, but great find!

I was running into this exact issue and had not seen this solution suggested elsewhere. Having read over the documentation again, the way this is stated can be confusing:
> ... you can declare any additional middleware you want Livewire to use ...

I read this as I should only add middleware if I need something in addition to the `web` middleware group, not that I need to include the `web` middleware group.

**Reply by joshhanley** [reply]
@serge-kilimoff and @justindantzer yep makes sense. PRs to update the docs are welcome! 🙂
